internal/deploy/apply.go

```diff
@@ -21,28 +22,30 @@ type applyOpts struct {
 	initialBackoff time.Duration
 }
 
-func applyManifests(ctx context.Context, k8s kube.KymaKube, manifests []byte, opts applyOpts) error {
+func applyManifests(ctx context.Context, k8s kube.KymaKube, manifests []byte, opts applyOpts) ([]client.Object, error) {
```

Comment:
Can you please split the actual "incremental retry apply" and the parsing of of bytes to manifestObjs? So you can patch the deployment obj in memory already?